### PR TITLE
Aca 60

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ACA-60

### Change description ###

Streamline the steps required to get ccd-docker up and running with a correctly configured SIDAM.

All manual steps for IDAM configuration should now be replaced with scripts, and README updated accordingly

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -32,3 +32,9 @@ ES_ENABLED_DOCKER=false
 #uncomment to enable Idam Stub
 #IDAM_STUB_SERVICE_NAME=http://ccd-test-stubs-service:5555
 #IDAM_STUB_LOCALHOST=http://localhost:5555
+
+#XUI WEBAPP Oauth2 secrets
+BEFTA_S2S_CLIENT_SECRET_OF_XUI_WEBAPP=OOOOOOOOOOOOOOOO
+
+#ACA Oauth2 secrets
+BEFTA_S2S_CLIENT_SECRET_OF_AAC_MANAGE_CASE_ASSIGNMENT=AAAAAAAAAAAAAAAA
```
Comment (MSancaktutar) [inline .env:40]: eol at eof pls

File: bin/users.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  {"email": "master.solicitor.3@gmail.com", "roles": "caseworker,caseworker-befta_master,caseworker-befta_master-solicitor_3", "lastName": "befta", "firstName": "solicitor3"},
+
+  {"email": "befta.master.solicitor.becky@gmail.com", "roles": "caseworker,caseworker-befta_master,caseworker-befta_master-solicitor", "lastName": "Solicington", "firstName": "Becky"},
+  {"email": "befta.master.solicitor.benjamin@gmail.com", "roles": "caseworker,caseworker-befta_master,caseworker-befta_master-solicitor", "lastName": "Solicington", "firstName": "Benjamin"},
+  {"email": "befta.master.solicitor.bill@gmail.com", "roles": "caseworker,caseworker-befta_master,caseworker-befta_master-solicitor", "lastName": "Solicington", "firstName": "Bill"},
+  {"email": "befta.master.solicitor.emma@gmail.com", "roles": "caseworker,caseworker-befta_master,caseworker-befta_master-solicitor", "lastName": "Solicington", "firstName": "Emma"},
+
+  {"email": "befta.caseworker.caa@gmail.com", "roles": "caseworker,caseworker-caa", "lastName": "CAA", "firstName": "Caseworker"}
+]
```
Comment (MSancaktutar) [inline bin/users.json:46]: eol at eof please

File: bin/utils/idam-add-role.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+}')
+
+if [ $STATUS -eq 201 ]; then
+  echo "Role created sucessfully"
+elif [ $STATUS -eq 409 ]; then
+  echo "Role already exists!"
+else
+  echo "ERROR: HTTPCODE = $STATUS"
+  exit 1
+fi
```
Comment (MSancaktutar) [inline bin/utils/idam-add-role.sh:33]: eol

File: bin/utils/idam-authenticate.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+set -eu
+
+USERNAME=$1
+PASSWORD=$2
+
+curl --silent --show-error --header 'Content-Type: application/x-www-form-urlencoded' -H 'Accept: application/json' -d "username=${USERNAME}&password=${PASSWORD}" "${IDAM_API_BASE_URL:-http://localhost:5000}/loginUser" | docker run --rm --interactive stedolan/jq -r .api_auth_token
```
Comment (MSancaktutar) [inline bin/utils/idam-authenticate.sh:8]: eol

**Review by MSancaktutar — APPROVED** [review]
LGTM, but will be good to have a second pair of eyes on this, especially because this will take effect for all CCD setup.

File: bin/utils/idam-create-service.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+set -eu
+
+dir=$(dirname ${0})
+
+LABEL=${1}
+CLIENT_ID=${2}
+CLIENT_SECRET=${3}
+REDIRECT_URL=${4}
+SELF_REGISTRATION=${5:-"false"}
+SCOPE=${6:-"openid profile authorities acr roles search-user"}
```
Comment (smathangi) [inline bin/utils/idam-create-service.sh:12]: please change default to "openid profile roles" to inline with IDAM.

File: README.md [hunk 6]
```diff
@@ -80,129 +80,260 @@ Mac : `source ./bin/set-environment-variables.sh`
...
+
+to add a new entry and re-run the script (any entries in this file that already exist are skipped)
+
+`${dir}/utils/idam-create-service.sh LABEL CLIENT_ID CLIENT_SECRET REDIRECT_URL SELF_REGISTRATION SCOPE`
+
+---
+**NOTE**
+
+* SELF_REGISTRATION - a boolean parameter, defaults to a value of "false" if omitted
+* SCOPE - a space delimited string parameter, defaults to a value of "openid profile authorities acr roles search-user" if omitted
```
Comment (smathangi) [inline README.md:184]: default should be "openid profile roles" (to inline with IDAM).

**Review by smathangi — APPROVED** [review]
LGTM (just one minor comment).  Very well done @sullivangareth 👏

File: .env (hunk 1 above)
Comment (sullivangareth) [reply .env]: Fixed in https://github.com/hmcts/ccd-docker/pull/103/commits/2a11ab59294f83f9ce45b5e51cda3ea2bc309264

File: bin/users.json (hunk 2 above)
Comment (sullivangareth) [reply bin/users.json]: https://github.com/hmcts/ccd-docker/pull/103/commits/2a11ab59294f83f9ce45b5e51cda3ea2bc309264

File: bin/utils/idam-add-role.sh (hunk 3 above)
Comment (sullivangareth) [reply bin/utils/idam-add-role.sh]: https://github.com/hmcts/ccd-docker/pull/103/commits/2a11ab59294f83f9ce45b5e51cda3ea2bc309264

File: bin/utils/idam-authenticate.sh (hunk 4 above)
Comment (sullivangareth) [reply bin/utils/idam-authenticate.sh]: https://github.com/hmcts/ccd-docker/pull/103/commits/2a11ab59294f83f9ce45b5e51cda3ea2bc309264

File: README.md (hunk 6 above)
Comment (sullivangareth) [reply README.md]: https://github.com/hmcts/ccd-docker/pull/103/commits/2a11ab59294f83f9ce45b5e51cda3ea2bc309264

File: bin/utils/idam-create-service.sh (hunk 5 above)
Comment (sullivangareth) [reply bin/utils/idam-create-service.sh]: https://github.com/hmcts/ccd-docker/pull/103/commits/2a11ab59294f83f9ce45b5e51cda3ea2bc309264

**Pull request merged**